contracts/src/v0.8/vrf/dev/VRFV2PlusWrapper.sol

```diff
@@ -340,20 +368,21 @@ contract VRFV2PlusWrapper is ConfirmedOwner, TypeAndVersionInterface, VRFConsume
 
   function _calculateRequestPriceNative(uint256 _gas, uint256 _requestGasPrice) internal view returns (uint256) {
     // costWei is the base fee denominated in wei (native)
-    // costWei takes into account the L1 posting costs of the VRF fulfillment
-    // transaction, if we are on an L2.
-    uint256 costWei = (_requestGasPrice *
-      (_gas + s_wrapperGasOverhead + s_coordinatorGasOverhead) +
-      ChainSpecificUtil._getL1CalldataGasCost(s_fulfillmentTxSizeBytes));
-    // ((wei/gas * (gas)) + l1wei)
-    // baseFee is the base fee denominated in wei
-    uint256 baseFee = costWei;
-    // feeWithPremium is the fee after the percentage premium is applied
-    uint256 feeWithPremium = (baseFee * (s_wrapperPremiumPercentage + 100)) / 100;
-    // feeWithFlatFee is the fee after the flat fee is applied on top of the premium
-    uint256 feeWithFlatFee = feeWithPremium + (1e12 * uint256(s_fulfillmentFlatFeeNativePPM));
-
-    return feeWithFlatFee;
+    // (wei/gas) * gas
+    uint256 wrapperCostWei = _requestGasPrice * s_wrapperGasOverhead;
+
+    // coordinatorCostWei takes into account the L1 posting costs of the VRF fulfillment transaction, if we are on an L2.
+    // (wei/gas) * gas + l1wei
+    uint256 coordinatorCostWei = _requestGasPrice *
+      (_gas + s_coordinatorGasOverhead) +
+      ChainSpecificUtil._getL1CalldataGasCost(s_fulfillmentTxSizeBytes);
+
+    // coordinatorCostWithPremiumAndFlatFeeWei is the coordinator cost with the percentage premium and flat fee applied
+    // coordinator cost * premium multiplier + flat fee
+    uint256 coordinatorCostWithPremiumAndFlatFeeWei = ((coordinatorCostWei * (s_wrapperNativePremiumPercentage + 100)) /
+      100) + (1e12 * uint256(s_fulfillmentFlatFeeNativePPM));
+
+    return wrapperCostWei + coordinatorCostWithPremiumAndFlatFeeWei;
```

Comment:
In the previous code, the following qty is added to the return value for s_wrapperGasOverhead
```
(_requestGasPrice * s_wrapperGasOverhead * (s_wrapperPremiumPercentage + 100)) / 100
```

After this change, only the following qty is added to return value for s_wrapperGasOverhead
```
_requestGasPrice * s_wrapperGasOverhead
```

Is this intentional?

Reply:
I'm not quite sure if I'm following you. But yes, we removed the premium for the wrapper itself and will only keep the premium for the wrapper.

Previously it's something like `((gas price * (wrapper gas + coordinator gas + consumer gas)) + L1 gas cost) * premium multiplier + flat fee`.

Now, after removing the premium and fees for wrapper, it becomes `(gas price * wrapper gas) + ((gas price * (coordinator gas + consumer gas)) + L1 gas cost) * premium mulitplier + flat fee`.

The only part I'm not sure is if we need the L1 gas cost for the wrapper part.